self-hosting/README.md

```diff
@@ -0,0 +1,304 @@
+
+## In detached mode 
+
+$ sudo docker-compose up  -d 
+
+##  Outside the config folder 
+
+$ sudo docker-compose -f /path/to/docker-compose.yml up -d
+
+```
+
+Note In certain shells, docker-compose may not interpolate the admin password that was exported above. In that case, your admin user had a password automatically generated. Note the New CouchDB Administrative User and New CouchDB Administrative Password in the output terminal. You can retrieve these via running `docker logs medic-os` and searching the terminal.
+
+Once containers are setup, please run the following command from your host terminal:
+
+```bash
+
+$ docker exec -it medic-os /bin/bash -c "sed -i 's/--install=3.9.0/--complete-install/g' /srv/scripts/horticulturalist/postrun/horticulturalist"
+$ docker exec -it medic-os /bin/bash -c "/boot/svc-disable medic-core openssh && /boot/svc-disable medic-rdbms && /boot/svc-disable medic-couch2pg"
+
+```
+
+The first command fixes a postrun script for horticulturalist to prevent unique scenarios of re-install. The second command removes extra services that you will not need.
+
+### Visit your project
+
+Open a browser to: https://localhost
+
+You will have to click to through the SSL Security warning. Click Advanced -> Continue to site.
+
+
+### Clean up an re-install
+
+You could have missed some of the instructions above and end with a bricked setup.  You can use the commands below to clean up and start afresh.
+
+Stop containers:
+
+- `docker-compose down or docker stop medic-os && docker stop haproxy`
+
+Remove containers:
+
+- `docker-compose rm or docker rm medic-os && docker rm haproxy`
+
+Clean data volume:
+
+- `docker volume rm medic-data`
+
+Note: Running `docker-compose down -v` from the docker-compose.yml directory would do all the above 3 steps
+
+Prune system:
+
+- `docker system prune`
+
+After following the above commands, you can re-run docker-compose up and create a fresh install (no previous data present)
+
+
+### Port Conflicts
+
+In case you are already running services on HTTP(80) and HTTPS(443), you will have to map new ports to the medic-os container.
+
+Turn off and remove all existing containers that were started:
+
+- `docker-compose down && docker-compose rm`
+To find out which service is using a conflicting port: On Linux:
+
+`sudo netstat -plnt | grep ':<port>'`
+On Mac (10.10 and above):
+
+`sudo lsof -iTCP -sTCP:LISTEN -n -P | grep ':<port>'` 
+
+You can either kill the service which is occupying HTTP/HTTPS ports, or run the container with forwarded ports that are free. In your compose file, change the ports under medic-os:
+
+```yaml
+
+services:
+  medic-os:
+    container_name: medic-os
+    image: medicmobile/medic-os:cht-3.7.0-rc.1
+    volumes:
+      - medic-data:/srv
+    ports:
+     - 8080:80
+     - 444:443
+
+```
+
+Note: You can  substitute 8080, 444 with whichever ports are free on your host. You would now visit https://localhost:444 to visit your project.
+
+### Helpful Docker Commands
```

Comment:
Some of these helpful commands below start with a `#` (eg `#docker ps`), some don't (eg `docker ps -f "status=exited"`).  We should standardize on one format and use it consistently.  With out I think?